File: tracker/src/test/java/testhelpers/QueryHashMap.java
package testhelpers;

import org.matomo.sdk.QueryParams;
import org.matomo.sdk.TrackMe;

import java.util.HashMap;


public class QueryHashMap extends HashMap<String, String> {

    public QueryHashMap(TrackMe trackMe) {
        super(trackMe.toMap());
    }

    public String get(QueryParams key) {
        return get(key.toString());
    }

}

File: tracker/src/test/java/testhelpers/QueryHashMap.kt
package testhelpers

import org.matomo.sdk.QueryParams
import org.matomo.sdk.TrackMe

class QueryHashMap(trackMe: TrackMe) : HashMap<String?, String?>(trackMe.toMap()) {
    fun get(key: QueryParams): String? {
        return get(key.toString())
    }
}
